### PR TITLE
[WEB-582] text input

### PR DESCRIPTION
**Pull request by krystophv**
See [WEB-582] for details

[WEB-582]: https://tidepool.atlassian.net/browse/WEB-582

### CODE REVIEW

File: app/components/elements/TextInput.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { Box } from 'rebass/styled-components';
+import { Label, Input } from '@rebass/forms';
+import { Caption } from './FontStyles';
+import { fonts, fontSizes, colors, borders, radii, space } from '../../themes/baseTheme';
+
+export const TextInput = props => {
+  const StyledInput = styled(Input)`
```
Comment (cbwebdevelopment) [inline app/components/elements/TextInput.js]: It's important, for performance reasons, to define the styled component outside of the render method or functional component that will be using it.

See https://styled-components.com/docs/basics#define-styled-components-outside-of-the-render-method for reference

File: app/components/elements/TextInput.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      sx={{
+        width: props.width,
+      }}
+    >
+      {props.label &&
+        <Label htmlFor={props.name}>
+          <Caption>{props.label}</Caption>
+        </Label>
+      }
+      <StyledInput id={props.name} {...props} />
```
Comment (cbwebdevelopment) [inline app/components/elements/TextInput.js]: This isn't a big deal, but I'm not sure if you intended to add the `label`, `name`, and `width` props to the `StyledInput` as well.  I don't think they could really cause issues (though I'm wondering if the width being added to the padding applied to the input could be...)

I'd consider pulling the other props out and getting the input props via rest destructuring (unless you intended to pass all the props through)

```
const { label, name, width, ...inputProps } = props;
...
<StyledInput id={props.name} {...inputProps} />
```

File: app/components/elements/TextInput.js (hunk 1 above)
Comment (krystophv) [reply app/components/elements/TextInput.js]: Should we also change it in `IconButton` then? (I used this as a reference to build this component out - https://github.com/tidepool-org/blip/blob/web-559-component-library/app/components/elements/IconButton.js#L8)

File: app/components/elements/TextInput.js (hunk 2 above)
Comment (krystophv) [reply app/components/elements/TextInput.js]: Not entirely intentional, but I didn't see it hurting so I shrugged it off and didn't come back to address it after time of writing. Destructured is much nicer/cleaner and more specific/intentional. 👍

File: app/components/elements/TextInput.js (hunk 1 above)
Comment (cbwebdevelopment) [reply app/components/elements/TextInput.js]: Ah yes!  I'll change it there.  Thanks for pointing that out!

**Pull request merged**